File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    arr=[]
    for i in range(1, 6):
        for x in product(["A", "E", "I", "O", "U"], repeat=i):
            arr.append(''.join(x))
    arr.sort()

    
    answer = arr.index(word)+1
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;
class Solution {
    static int cnt;
    static String w;
    static boolean flag;
    static int answer;
    static String arr[] = {"A","E","I","O","U"};
    public int solution(String word) {
        answer = 0;
        w = word;
        dfs("");
        return answer;
    }
    static void dfs(String s) {
        if(flag) return;
        if(s.length()>0) {
            cnt++;
            System.out.println(s);
            if(s.equals(w)) {
                flag = true;
                answer = cnt;
                return;
            }
        }
        if(s.length()==5) return;
        for(int i=0; i<5; i++) {
            dfs(s+arr[i]);
            if(flag) return;
        }
    }
}